code/modules/mob/living/carbon/human/human.dm

```diff
@@ -40,6 +40,14 @@
 
 	UpdateAppearance()
 	GLOB.human_list += src
+	if(isskrell(src) || isgrey(src) || ismachineperson(src) || iskidan(src) || iswryn(src) || isabductor(src) || ishumanbasic(src) || isdiona(src) || isplasmaman(src) || isskeleton(src))
+		AddComponent(/datum/component/footstep, FOOTSTEP_MOB_HUMAN, 1, 2)
+	if(istajaran(src) || isunathi(src) || isvulpkanin(src))
+		AddComponent(/datum/component/footstep, FOOTSTEP_MOB_CLAW, 1, 2)
+	if(isdrask(src) || isvoxarmalis(src))
+		AddComponent(/datum/component/footstep, FOOTSTEP_MOB_HEAVY, 1, 2)
+	if(isslimeperson(src))
+		AddComponent(/datum/component/footstep, FOOTSTEP_MOB_SLIME, 1, 2)
```

Comment:
You might consider adding a species variable, named `footstep_type` for example, which will hold a footstep define (`FOOTSTEP_MOB_HUMAN`, etc.). Then you can do:
```dm
AddComponent(/datum/component/footstep, dna.species.footstep_type, 1, 2)
```

Reply:
Yessssss! Great idea! :O